File: backend/app/schemas/action_log.py
from datetime import datetime

from pydantic import BaseModel


class ActionLogOut(BaseModel):
    id: int
    actor_id: int
    actor_name: str
    action: str
    entity_type: str
    entity_id: int
    details: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/actionLog.ts
export interface ActionLogEntry {
  id: number
  actor_id: number
  actor_name: string
  action: string
  entity_type: string
  entity_id: number
  details: string
  created_at: string
}

export async function fetchActionLog(params?: { entity_type?: string; actor_id?: number }): Promise<ActionLogEntry[]> {
  const search = new URLSearchParams()
  if (params?.entity_type) search.set('entity_type', params.entity_type)
  if (params?.actor_id != null) search.set('actor_id', String(params.actor_id))
  const query = search.toString()
  const res = await fetch(`/api/action-log/${query ? `?${query}` : ''}`, { credentials: 'include' })
  if (!res.ok) throw new Error('Failed to fetch action log')
  return res.json()
}
